cpp/src/arrow/c/abi.h

```diff
@@ -60,6 +60,31 @@ struct ArrowArray {
   void* private_data;
 };
 
+// EXPERIMENTAL
+struct ArrowArrayStream {
+  // Callback to get the stream type
+  // (will be the same for all arrays in the stream).
+  // Return value: 0 if successful, an `errno`-compatible error code otherwise.
+  int (*get_schema)(struct ArrowArrayStream*, struct ArrowSchema* out);
+  // Callback to get the next array
+  // (if no error and the array is released, the stream has ended)
```

Comment:
`released` is something defined in the ArrowArray-spec.  Is it stronger or weaker guarantee then returning a nullptr here?

Reply:
A nullptr cannot be returned. The callback returns an `int`. However, we could say that returning `-1` means end of stream.